ouroboros-consensus/src/Ouroboros/Consensus/Protocol/PBFT/State.hs

```diff
@@ -360,155 +202,35 @@ tipSlot cs@PBftState{..} = lastSignedSlot cs `max` ebbsMax ebbs
 -- In other words, the PBFT chain state corresponding to genesis.
 empty :: PBftState c
 empty = PBftState {
-      preAnchor  = Empty
-    , postAnchor = Empty
-    , preWindow  = Empty
-    , inWindow   = Empty
+      inWindow   = Empty
     , counts     = Map.empty
-    , ebbs       = ebbsEmpty
     }
 
 -- | Append new signature
 --
 -- Drops the oldest signature, provided we have reached the required number.
-append :: forall c. PBftCrypto c
-       => SecurityParam
-       -> WindowSize
-       -> PBftSigner c
-       -> PBftState c -> PBftState c
-append k n signer@(PBftSigner _ gk) PBftState{..} =
-    assertInvariant k n $
-    pruneEBBsLT $
-    PBftState {
-        preAnchor  = preAnchor'
-      , postAnchor = postAnchor'
-      , preWindow  = preWindow'
-      , inWindow   = inWindow'
+append ::
+     forall c. PBftCrypto c
+  => WindowSize
+  -> PBftSigner c
+  -> PBftState c -> PBftState c
+append n signer@(PBftSigner _ gk) PBftState{..} =
```

Comment:
Minor: this might be more obviously correct if we split it into a grow step and shrink step.

Reply:
Done